### DISCUSSION TITLE
Can create non-namespaced resources error

### DISCUSSION THREAD

**Discussion by CoreyCBurton** [linkerd/linkerd2#6025] [Q&A] [answered]
After following the guide on linkerd.com to install it on my K3sup Cluster server, I have run into a few problems. Bear with me if this is an easy fix, I am still learning how clusters work and Linkerd.

My environment is Ubuntu Server LTS 20.04.2 on a Raspberry Pi4 which has K3sup installed. I have set up the nodes properly and when I pull my nodes in the terminal, this is what it shows. 

```
NAME      STATUS   ROLES    AGE   VERSION        INTERNAL-IP     EXTERNAL-IP   OS-IMAGE             KERNEL-VERSION     CONTAINER-RUNTIME
delta     Ready    <none>   35h   v1.19.9+k3s1   192.168.1.39    <none>        Ubuntu 20.04.2 LTS   5.4.0-1032-raspi   containerd://1.4.4-k3s1
alpha     Ready    master   36h   v1.19.9+k3s1   192.168.1.98    <none>        Ubuntu 20.04.2 LTS   5.4.0-1032-raspi   containerd://1.4.4-k3s1
beta      Ready    <none>   36h   v1.19.9+k3s1   192.168.1.143   <none>        Ubuntu 20.04.2 LTS   5.4.0-1032-raspi   containerd://1.4.4-k3s1
chairle   Ready    <none>   35h   v1.19.9+k3s1   192.168.1.19    <none>        Ubuntu 20.04.2 LTS   5.4.0-1028-raspi   containerd://1.4.4-k3s1
```

I have installed everything and when I run `linked check`, I get one x on the buoyant extension which is shown below 
```
× buoyant-cloud-metrics Deployment is running
    container buoyant-cloud-metrics-774b5b4898-8qf8s in pod linkerd-proxy is not ready 
```
Running through` linkerd check --pre --linkerd-namespace linkerd-test `I have also discovered this problem
```
 × can create non-namespaced resources
    cannot create ClusterRole/linkerd-heartbeat: clusterroles.rbac.authorization.k8s.io "linkerd-heartbeat" already exists
    cannot create ClusterRoleBinding/linkerd-heartbeat: clusterrolebindings.rbac.authorization.k8s.io "linkerd-heartbeat" already exists
    cannot create CustomResourceDefinition/serviceprofiles.linkerd.io: customresourcedefinitions.apiextensions.k8s.io "serviceprofiles.linkerd.io" already exists
    cannot create CustomResourceDefinition/trafficsplits.split.smi-spec.io: customresourcedefinitions.apiextensions.k8s.io "trafficsplits.split.smi-spec.io" already exists
    cannot create MutatingWebhookConfiguration/linkerd-proxy-injector-webhook-config: mutatingwebhookconfigurations.admissionregistration.k8s.io "linkerd-proxy-injector-webhook-config" already exists
    cannot create ValidatingWebhookConfiguration/linkerd-sp-validator-webhook-config: validatingwebhookconfigurations.admissionregistration.k8s.io "linkerd-sp-validator-webhook-config" already exists
    see https://linkerd.io/checks/#pre-k8s-cluster-k8s for hints
```
When I run `linkerd viz dashboard &`

```
http://localhost:50750
Grafana dashboard available at:
http://localhost:50750/grafana
Opening Linkerd dashboard in the default browser
Failed to open Linkerd dashboard automatically
Visit http://localhost:50750 in your browser to view the dashboard
```
When I open the link on my machine, the site cannot be reached. What is the solution to this problem? Thank you.

**Comment by mateiidavid** [comment]
Hey @CoreyCBurton 👋🏻 ,

The errors you are seeing with `linkerd check --pre` are to be expected. The --pre check is used prior to installing Linkerd, it complains that certain cluster scoped resources (e.g crds such as serviceprofiles and trafficsplits) are already installed. If you'd uninstall Linkerd and run the --pre checks again they would pass.

The error you are seeing with `linkerd viz dashboard` means that you can't be redirected to your browser. My assumption here is that you're working on a remote machine, over SSH -- in this case, you won't be redirected in your local machine's browser.

If the assumption here is correct then the fix is pretty simple: you can either set up a port-forward (e.g [LocalForward](https://www.ssh.com/academy/ssh/tunneling/example) 8084 over ssh so you can connect to localhost:8084 on your local machine and then port-forward the pod) or set up an ingress.

Let me know what you think and if this helps you out :)

**Comment by CoreyCBurton** [comment, marked as answer]
There is a solution to this problem. I was opening `linkerd viz dashboard &` on my master node. I exported my kubeconfig on my local machine and install Linkerd there. I was able to connect to the http://localhost:50750.
